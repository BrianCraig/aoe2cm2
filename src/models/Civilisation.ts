import GameVersion from "./GameVersion";

enum Name {
    ARABIA = "Arabia",
    ARENA = "Arena",
    CENOTES = "Cenotes",
    GOLDEN_PIT = "Golden Pit",
    HIDEOUT = "Hideout",
    ISLANDS = "Islands",
    MEDITERRANEAN = "Mediterranean",
    SERENGETI = "Serengeti",
    VALLEY = "Valley",
    CROSS = "Cross",
    FORTRESS = "Fortress",
    GRAVEYARDS = "Graveyards",
    LAND_MADNESS = "Land Madness",

    HIDDEN_PICK = "HIDDEN_PICK",
    HIDDEN_BAN = "HIDDEN_BAN",
    HIDDEN_SNIPE = "HIDDEN_SNIPE",
    HIDDEN = "HIDDEN",
    RANDOM = "RANDOM"
}

class Civilisation {

    public static readonly ARABIA:Civilisation = new Civilisation(Name.ARABIA, GameVersion.DEFINITIVE_EDITION);
    public static readonly ARENA:Civilisation = new Civilisation(Name.ARENA, GameVersion.DEFINITIVE_EDITION);
    public static readonly CENOTES:Civilisation = new Civilisation(Name.CENOTES, GameVersion.DEFINITIVE_EDITION);
    public static readonly GOLDEN_PIT:Civilisation = new Civilisation(Name.GOLDEN_PIT, GameVersion.DEFINITIVE_EDITION);
    public static readonly HIDEOUT:Civilisation = new Civilisation(Name.HIDEOUT, GameVersion.DEFINITIVE_EDITION);
    public static readonly ISLANDS:Civilisation = new Civilisation(Name.ISLANDS, GameVersion.DEFINITIVE_EDITION);
    public static readonly MEDITERRANEAN:Civilisation = new Civilisation(Name.MEDITERRANEAN, GameVersion.DEFINITIVE_EDITION);
    public static readonly SERENGETI:Civilisation = new Civilisation(Name.SERENGETI, GameVersion.DEFINITIVE_EDITION);
    public static readonly VALLEY:Civilisation = new Civilisation(Name.VALLEY, GameVersion.DEFINITIVE_EDITION);
    public static readonly CROSS:Civilisation = new Civilisation(Name.CROSS, GameVersion.DEFINITIVE_EDITION);
    public static readonly FORTRESS:Civilisation = new Civilisation(Name.FORTRESS, GameVersion.DEFINITIVE_EDITION);
    public static readonly GRAVEYARDS:Civilisation = new Civilisation(Name.GRAVEYARDS, GameVersion.DEFINITIVE_EDITION);
    public static readonly LAND_MADNESS:Civilisation = new Civilisation(Name.LAND_MADNESS, GameVersion.DEFINITIVE_EDITION);

    public static readonly RANDOM:Civilisation = new Civilisation(Name.RANDOM, GameVersion.DEFINITIVE_EDITION);

    public static readonly HIDDEN_PICK: Civilisation = new Civilisation(Name.HIDDEN_PICK, GameVersion.TECHNICAL);
    public static readonly HIDDEN_BAN: Civilisation = new Civilisation(Name.HIDDEN_BAN, GameVersion.TECHNICAL);
    public static readonly HIDDEN_SNIPE: Civilisation = new Civilisation(Name.HIDDEN_SNIPE, GameVersion.TECHNICAL);
    public static readonly HIDDEN: Civilisation = new Civilisation(Name.HIDDEN, GameVersion.TECHNICAL);

    // DO NOT CHANGE THE ORDER OF ELEMENTS IN THIS ARRAY!!!
    // ONLY APPEND NEW CIVILISATIONS AT THE END!!!
    public static readonly ALL = [
        Civilisation.ARABIA,
        Civilisation.ARENA,
        Civilisation.CENOTES,
        Civilisation.GOLDEN_PIT,
        Civilisation.HIDEOUT,
        Civilisation.ISLANDS,
        Civilisation.MEDITERRANEAN,
        Civilisation.SERENGETI,
        Civilisation.VALLEY,
        Civilisation.CROSS,
        Civilisation.FORTRESS,
        Civilisation.GRAVEYARDS,
        Civilisation.LAND_MADNESS
    ];

    public readonly name:Name;
    public readonly gameVersion:GameVersion;
    public isRandomlyChosenCiv: boolean = false;

    private constructor(name:Name, gameVersion:GameVersion){
        this.name = name;
        this.gameVersion = gameVersion;
    }

}

export default Civilisation;
