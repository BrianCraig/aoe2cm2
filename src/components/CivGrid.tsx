import * as React from 'react';
import '../pure-min.css'
import '../style2.css'
import Civilisation from "../models/Civilisation";
import CivPanel from "../containers/CivPanel";
import ActionType from "../models/ActionType";
import Socket = SocketIOClient.Socket;

interface IProps {
    civilisations: Civilisation[],
    socket: Socket
}

class CivGrid extends React.Component<IProps, object> {
    public render() {

        const civPanels = this.props.civilisations.map((civ, index) => {
            return React.createElement(CivPanel, {actionType: ActionType.CHOICE, active: false, civilisation: civ, socket: this.props.socket});
        });

        return (
            <div>
                <div id="civgrid" className="chooser card">
                    <div className="pure-g chooser-grid box-content">
                        {civPanels}
                    </div>
                    <div className="card-background">cbg</div>
                </div>
            </div>
        );
    }
}

export default CivGrid;
