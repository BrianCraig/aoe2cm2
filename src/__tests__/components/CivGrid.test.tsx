import {shallow} from "enzyme";
import * as React from "react";
import CivGrid from "../../components/draft/CivGrid";
import Civilisation from "../../models/Civilisation";

it('CivGrid renders correctly', () => {
    const component = shallow(<CivGrid civilisations={Civilisation.ALL}/>);
    expect(component).toMatchSnapshot();
});
