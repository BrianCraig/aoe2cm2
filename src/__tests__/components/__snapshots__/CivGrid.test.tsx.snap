// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CivGrid renders correctly 1`] = `
<div>
  <div
    className="chooser card"
    id="civgrid"
  >
    <div
      className="pure-g chooser-grid box-content"
    >
      <Connect(CivPanel)
        active={false}
        civPanelType="choice"
        civilisation={
          Civilisation {
            "gameVersion": 2,
            "name": "Aztecs",
          }
        }
      />
      <Connect(CivPanel)
        active={false}
        civPanelType="choice"
        civilisation={
          Civilisation {
            "gameVersion": 4,
            "name": "Berbers",
          }
        }
      />
      <Connect(CivPanel)
        active={false}
        civPanelType="choice"
        civilisation={
          Civilisation {
            "gameVersion": 1,
            "name": "Britons",
          }
        }
      />
      <Connect(CivPanel)
        active={false}
        civPanelType="choice"
        civilisation={
          Civilisation {
            "gameVersion": 5,
            "name": "Burmese",
          }
        }
      />
      <Connect(CivPanel)
        active={false}
        civPanelType="choice"
        civilisation={
          Civilisation {
            "gameVersion": 1,
            "name": "Byzantines",
          }
        }
      />
      <Connect(CivPanel)
        active={false}
        civPanelType="choice"
        civilisation={
          Civilisation {
            "gameVersion": 1,
            "name": "Celts",
          }
        }
      />
      <Connect(CivPanel)
        active={false}
        civPanelType="choice"
        civilisation={
          Civilisation {
            "gameVersion": 1,
            "name": "Chinese",
          }
        }
      />
      <Connect(CivPanel)
        active={false}
        civPanelType="choice"
        civilisation={
          Civilisation {
            "gameVersion": 4,
            "name": "Ethiopians",
          }
        }
      />
      <Connect(CivPanel)
        active={false}
        civPanelType="choice"
        civilisation={
          Civilisation {
            "gameVersion": 1,
            "name": "Franks",
          }
        }
      />
      <Connect(CivPanel)
        active={false}
        civPanelType="choice"
        civilisation={
          Civilisation {
            "gameVersion": 1,
            "name": "Goths",
          }
        }
      />
      <Connect(CivPanel)
        active={false}
        civPanelType="choice"
        civilisation={
          Civilisation {
            "gameVersion": 2,
            "name": "Huns",
          }
        }
      />
      <Connect(CivPanel)
        active={false}
        civPanelType="choice"
        civilisation={
          Civilisation {
            "gameVersion": 3,
            "name": "Incas",
          }
        }
      />
      <Connect(CivPanel)
        active={false}
        civPanelType="choice"
        civilisation={
          Civilisation {
            "gameVersion": 3,
            "name": "Indians",
          }
        }
      />
      <Connect(CivPanel)
        active={false}
        civPanelType="choice"
        civilisation={
          Civilisation {
            "gameVersion": 3,
            "name": "Italians",
          }
        }
      />
      <Connect(CivPanel)
        active={false}
        civPanelType="choice"
        civilisation={
          Civilisation {
            "gameVersion": 1,
            "name": "Japanese",
          }
        }
      />
      <Connect(CivPanel)
        active={false}
        civPanelType="choice"
        civilisation={
          Civilisation {
            "gameVersion": 5,
            "name": "Khmer",
          }
        }
      />
      <Connect(CivPanel)
        active={false}
        civPanelType="choice"
        civilisation={
          Civilisation {
            "gameVersion": 2,
            "name": "Koreans",
          }
        }
      />
      <Connect(CivPanel)
        active={false}
        civPanelType="choice"
        civilisation={
          Civilisation {
            "gameVersion": 3,
            "name": "Magyars",
          }
        }
      />
      <Connect(CivPanel)
        active={false}
        civPanelType="choice"
        civilisation={
          Civilisation {
            "gameVersion": 5,
            "name": "Malay",
          }
        }
      />
      <Connect(CivPanel)
        active={false}
        civPanelType="choice"
        civilisation={
          Civilisation {
            "gameVersion": 4,
            "name": "Malians",
          }
        }
      />
      <Connect(CivPanel)
        active={false}
        civPanelType="choice"
        civilisation={
          Civilisation {
            "gameVersion": 2,
            "name": "Mayans",
          }
        }
      />
      <Connect(CivPanel)
        active={false}
        civPanelType="choice"
        civilisation={
          Civilisation {
            "gameVersion": 1,
            "name": "Mongols",
          }
        }
      />
      <Connect(CivPanel)
        active={false}
        civPanelType="choice"
        civilisation={
          Civilisation {
            "gameVersion": 1,
            "name": "Persians",
          }
        }
      />
      <Connect(CivPanel)
        active={false}
        civPanelType="choice"
        civilisation={
          Civilisation {
            "gameVersion": 4,
            "name": "Portuguese",
          }
        }
      />
      <Connect(CivPanel)
        active={false}
        civPanelType="choice"
        civilisation={
          Civilisation {
            "gameVersion": 1,
            "name": "Saracens",
          }
        }
      />
      <Connect(CivPanel)
        active={false}
        civPanelType="choice"
        civilisation={
          Civilisation {
            "gameVersion": 3,
            "name": "Slavs",
          }
        }
      />
      <Connect(CivPanel)
        active={false}
        civPanelType="choice"
        civilisation={
          Civilisation {
            "gameVersion": 2,
            "name": "Spanish",
          }
        }
      />
      <Connect(CivPanel)
        active={false}
        civPanelType="choice"
        civilisation={
          Civilisation {
            "gameVersion": 1,
            "name": "Teutons",
          }
        }
      />
      <Connect(CivPanel)
        active={false}
        civPanelType="choice"
        civilisation={
          Civilisation {
            "gameVersion": 1,
            "name": "Turks",
          }
        }
      />
      <Connect(CivPanel)
        active={false}
        civPanelType="choice"
        civilisation={
          Civilisation {
            "gameVersion": 5,
            "name": "Vietnamese",
          }
        }
      />
      <Connect(CivPanel)
        active={false}
        civPanelType="choice"
        civilisation={
          Civilisation {
            "gameVersion": 1,
            "name": "Vikings",
          }
        }
      />
    </div>
    <div
      className="card-background"
    >
      cbg
    </div>
  </div>
</div>
`;
